### ISSUE TITLE
PyTorch sampler is mutually exclusive with shuffle 

### ISSUE THREAD

**Issue by whoisjones** [flairNLP/flair#2737]
**Describe the bug**
Using default settings, a custom sampler will raise an error since shuffle and a custom sampler a mutually exclusive in torch.

**Expected behavior**
DataLoader needs to check whether a sampler is given and thus overwrite shuffle default param.
